src/Kuery.ts

```diff
@@ -0,0 +1,77 @@
+import { drop, flow, keys, map, orderBy, take } from 'lodash/fp';
+import QueryCompiler, { Query } from './QueryCompiler';
+
+export interface KueryOptions {
+  skip: number;
+  limit: number;
+  sort: Record<string, number>;
```

Comment:
Why are these specified as mandatory? Based on usage, they appear to be optional. Then the type casting in the constructor can be removed as well.

Reply:
Solved by adding `protected options: Partial<KueryOptions>;`

Reply:
Why protected?